src/useFetchyeContext.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { useContext, useReducer } from 'react';
+import { FetchyeContext } from './FetchyeContext';
+import { defaultFetcher as libraryFetcher } from './defaultFetcher';
+import SimpleCache from './cache/SimpleCache';
+
+export const useFetchyeContext = (fallbacks = {}) => {
+  const {
+    fallbackFetchClient = fetch,
+    fallbackCache = SimpleCache(),
+    fallbackFetcher = libraryFetcher,
+  } = fallbacks;
+  // Setup headless mode fallbacks
+  const [state, fallbackDispatch] = useReducer(fallbackCache.reducer, fallbackCache.reducer(undefined, { type: '' }));
+
+  const fallbackUseFetchyeSelector = (hash) => ({
+    current: fallbackCache.getCacheByKey(state, hash),
+  });
+
+  const providedContext = useContext(FetchyeContext);
+  const {
+    cache = fallbackCache,
+    defaultFetcher = fallbackFetcher,
+    useFetchyeSelector = fallbackUseFetchyeSelector,
+    dispatch = fallbackDispatch,
+    fetchClient = fallbackFetchClient,
+  } = providedContext || {};
+
+  return {
+    cache,
+    defaultFetcher,
+    useFetchyeSelector,
+    dispatch,
+    fetchClient,
+    headless: true,
+  };
```

Comment:
Do you think we should wrap this in `useMemo`?

Reply:
No. Not worth it there are no computationally expensive values to save here. Here's an article that describes pragmatism and prevention of premature optimization with regards to `useMemo` and `useCallback`: https://kentcdodds.com/blog/usememo-and-usecallback